File: ShapesPainter/app/src/main/java/com/example/alexander/shapespainter/Ellipse.java
package com.example.alexander.shapespainter;

import android.graphics.Color;

/**
 * Created by Alexander on 20.01.2017.
 */

public class Ellipse extends Shape {

    public Ellipse(Vector2f center, VectorSize2f size, Color fillColor, Color outlineColor) {

    }

    @Override
    void draw(ICanvas canvas) {

    }

    @Override
    public ShapeType getType() {
        return null;
    }

    @Override
    public ShapeDiagram getDiagram() {
        return null;
    }

    @Override
    public Color getFillColor() {
        return null;
    }

    @Override
    public Color getOutlineColor() {
        return null;
    }

    @Override
    public Vector2f getVertices() {
        return null;
    }

    @Override
    public boolean isPointInside(Vector2f point) {
        return false;
    }

    @Override
    public Vector2f getCenter() {
        return null;
    }

    @Override
    public Vector2f getSize() {
        return null;
    }
}

File: Fabric/app/src/main/java/com/example/alexander/fabric/DesignerTest/Ellipse.kt
package com.example.alexander.fabric.DesignerTest

import com.example.alexander.fabric.Definitions.ColorEnum
import com.example.alexander.fabric.Definitions.CordPoint
import com.example.alexander.fabric.Interfaces.ICanvas
import com.example.alexander.fabric.Interfaces.Shape

/**
 * Created by Alexander on 25.06.2016.
 */
class Ellipse(color: ColorEnum, var center: CordPoint, var hRadius: Float, var wRadius: Float): Shape(color)
{
    override fun Draw(canvas: ICanvas)
    {
        canvas.SetColor(color)
        canvas.DrawEllipse(center, hRadius, wRadius)
    }
    override fun toString(): String
    {
        return "Ellipse: color: ${color.name} data: ${center.toString()}, ${hRadius.toString()}, ${wRadius.toString()}"
    }
}
